### ISSUE TITLE
cuda compilation error

### ISSUE THREAD

**Issue by PerspectivesLab** [DavidXu-JJ/PoissonRecon_GPU#2]
hi there, compiling on windows using cuda 11.8: 

expression must have a constant value main.cu	line : 3339	
ConfirmedPPolynomial<convTimes + 1, convTimes + 2> baseFunctions_h[fData.res];

 the size of an array must be greater than zero  PPolynomial.cuh	line :178	
 StartingPolynomial<Degree> *h_addr=pp_h->polys;
 
 any solution to resolve these errors ? thx

**Comment by DavidXu-JJ** [comment]
I tested this code in cuda 11.7 and ubuntu 20.04. And the code ran well when I developed it last year.

The first expression you mentioned in my code seems to be my mistake. And I'm also confused why it didn't report an error when I was actively developing this code.

The second expression you mentioned in my code seems to assign a pointer in cpu memory to another pointer. I'm not sure what is the `size of an array` you mention.

Current I'm developing another project and I will be busy for one month. Sorry I haven't dug into your problem deeply right now.

 If you are not urgent to solve this issue, I will try to handle this issue after I finish my current project.
